crates/oxc_linter/src/rules/eslint/no_compare_neg_zero.rs

```diff
@@ -110,6 +110,14 @@ impl Rule for NoCompareNegZero {
             }
         }
     }
+
+    fn should_run(&self, ctx: &crate::context::ContextHost) -> bool {
+        ctx.semantic().nodes().contains(oxc_ast::AstType::BinaryExpression)
+            && ctx
+                .semantic()
+                .nodes()
+                .contains_any(&[oxc_ast::AstType::NumericLiteral, oxc_ast::AstType::BigIntLiteral])
+    }
```

Comment:
From benchmarks, obviously some of these changes are having a positive effect.

But some of these methods may be of limited value. How many ASTs don't contain a single `BinaryExpression` and not a single `NumericLiteral`?

I imagine most of the gain is from e.g. `AstType::Class` and `AstType::DebuggerStatement`.

You may want to "pick your fights".